ekncontent/ekncontent/eknc-domain.c

```diff
@@ -294,24 +316,39 @@ eknc_domain_initable_init (GInitable *initable,
   EkncDomain *self = EKNC_DOMAIN (initable);
   GError *local_error = NULL;
 
-  if (self->app_id == NULL || *self->app_id == '\0')
+  gboolean has_app_id = (self->app_id != NULL && *self->app_id != '\0');
+  gboolean has_path = (self->path != NULL && *self->path != '\0');
+
+  g_autoptr(GFile) subscriptions_dir = NULL;
```

Comment:
You can move this variable and bundle_subscriptions_dir under `else if (has_app_id)`